### PR TITLE
updated to fint-buildscripts v1.4.0

### PR DESCRIPTION
**Pull request by jarlehansen**
Some minor changes, also added a few TODOs.
Can you have a quick look @asgeirn to see if this can be merged?

### CODE REVIEW

File: src/main/java/no/fint/provider/events/admin/AdminService.java [hunk 1]
```diff
@@ -28,26 +28,27 @@
 
     private RestTemplate restTemplate = new RestTemplate();
 
-    @Value("${fint.provider.assets.endpoint:}")
+    @Value("${fint.provider.assets.endpoint:}") // TODO move to ProviderProps?
     private String assetsEndpoint;
 
-    @Value("${fint.events.orgIds:}")
+    @Value("${fint.events.orgIds:}") // TODO rename to fint.provider.events.orgIds? Move to ProviderProps?
```
Comment (asgeirn) [inline src/main/java/no/fint/provider/events/admin/AdminService.java]: The property name is the same for consumer and provider, I'd suggest not remaining it.

File: src/main/java/no/fint/provider/events/admin/AdminService.java [hunk 2]
```diff
@@ -57,7 +58,7 @@ public boolean isRegistered(String orgId) {
     }
 
     public boolean register(String orgId, String client) {
-        if (!isRegistered(orgId) && ArrayUtils.isNotEmpty(validAssets) && Stream.of(validAssets).noneMatch(orgId::equals)) {
+        if (orgNotEnabled(orgId)) {
```
Comment (asgeirn) [inline src/main/java/no/fint/provider/events/admin/AdminService.java]: Suggest splitting `!isRegistered()` (which could be renamed to `notRegistered()`) and `orgNotEnabled()` - they are semantically two separate things.

File: src/main/java/no/fint/provider/events/eventstate/EventStateService.java [hunk 3]
```diff
@@ -36,14 +36,12 @@ public void add(Event event, int timeToLiveInMinutes) {
         eventStates.put(event.getCorrId(), new EventState(event, timeToLiveInMinutes));
     }
 
-    public Optional<EventState> get(Event event) {
-        Optional<EventState> result = Optional.ofNullable(eventStates.get(event.getCorrId()));
-        return result;
+    EventState get(Event event) {
```
Comment (asgeirn) [inline src/main/java/no/fint/provider/events/eventstate/EventStateService.java]: This method is now only used by `EventStateServiceSpec`.  Perhaps the specification should be rewritten?

File: src/main/java/no/fint/provider/events/admin/AdminService.java (hunk 2 above)
Comment (jarlehansen) [reply src/main/java/no/fint/provider/events/admin/AdminService.java]: sounds good

File: src/main/java/no/fint/provider/events/eventstate/EventStateService.java (hunk 3 above)
Comment (jarlehansen) [reply src/main/java/no/fint/provider/events/eventstate/EventStateService.java]: That is what I noticed as well, and why I changed it. I can look into it more.

File: src/main/java/no/fint/provider/events/admin/AdminService.java (hunk 1 above)
Comment (jarlehansen) [reply src/main/java/no/fint/provider/events/admin/AdminService.java]: Ok, good. Then we can remove the TODO

**Comment by asgeirn** [comment]
Thanks!  Moving `@Value` expressions to `ProviderProps` sounds like a good idea.

File: src/main/java/no/fint/provider/events/admin/AdminController.java [hunk 4]
```diff
@@ -39,22 +39,22 @@
     private AdminService adminService;
 
     @Autowired
-    private MongoTemplate mongoTemplate;
+    private MongoOperations mongo;
```
Comment (jarlehansen) [inline src/main/java/no/fint/provider/events/admin/AdminController.java]: @asgeirn fyi, this was the only problem I had when updating the spring boot version.By changing this to MongoOperations interface the tests passed.

Comment (asgeirn) [reply src/main/java/no/fint/provider/events/admin/AdminController.java]: Ah, good to know!

**Pull request merged**